lib/features/cli/networking/sync_session.dart

```diff
@@ -0,0 +1,23 @@
+import 'dart:async';
+
+import '../../../core/async/event_synchronizer.dart';
+import '../../../core/networking/websockets/session.dart';
+
+/// Creates an [EventSynchronizer] for a [ClientSession].
+///
+/// An optional [messageHandler] can be assigned which executes on
+/// every message received from the server.
+EventSynchronizer syncSession(
+  ClientSession session, [
+  FutureOr<void> Function(dynamic)? messageHandler,
+]) {
+  final sync = EventSynchronizer(session.send, Duration(milliseconds: 100));
```

Comment:
`sync` seems to be a [reserved keyword](https://jelenaaa.medium.com/what-are-sync-async-yield-and-yield-in-dart-defe57d06381) in Dart. Didn't know that either until GitHub helpfully colored it for me ^^